tests/unit/cocoon/test_cocoon_printer.py

```diff
@@ -0,0 +1,304 @@
+                            reason="not found"
+                        )
+                    ],
+                "success":
+                    [
+                        models.UpsertQuotesResponse(
+                            failed={
+                                "BBG001MM1KV4-Figi_2019-10-28":
+                                    models.Quote(
+                                        as_at="2019-01-16",
+                                        quote_id=models.QuoteId(
+                                            quote_series_id=models.QuoteSeriesId(
+                                                provider="Default",
+                                                instrument_id="BBG001MM1KV4",
+                                                instrument_id_type="Figi",
+                                                quote_type="Price",
+                                                field="Mid",
+                                            ),
+                                            effective_at="2019-01-16"
+                                        ),
+                                        uploaded_by="test"
+                                    )
+                            },
+                            values={
+                                "BBG001MM1KV4-Figi_2019-10-28":
+                                    models.Quote(
+                                        as_at="2019-01-16",
+                                        quote_id=models.QuoteId(
+                                            quote_series_id=models.QuoteSeriesId(
+                                                provider="Default",
+                                                instrument_id="BBG001MM1KV4",
+                                                instrument_id_type="Figi",
+                                                quote_type="Price",
+                                                field="Mid",
+                                            ),
+                                            effective_at="2019-01-16"
+                                        ),
+                                        uploaded_by="test"
+                                    )
+                            }
+
+                        )
+                    ]
+            },
+            "holdings": {
+                "errors": [
+                        lusid.exceptions.ApiException(
+                            status="404",
+                            reason="not found"
+                        )
+                    ],
+                "success": [
+                    models.AdjustHolding(
+                        href="https://notadomain.lusid.com/api/api/etc",
+                        version="1",
+                    )
+                ],
+            },
+        }
+        cls.responses_no_error_field = {
+            "instruments": {
+                "success": [
+                    models.UpsertInstrumentsResponse(
```

Comment:
Nice work testing the non-happy path here! :)

Reply:
Thanks!